### ISSUE TITLE
Access to `ILogEventSink`

### ISSUE THREAD

**Issue by droosma** [serilog/serilog-sinks-console#56]
**Does this issue relate to a new *feature* or an existing *bug*?**
- [ ] Bug
- [x] New Feature

**What version of Serilog Console Sink is affected by this issue?  Please list the related NuGet package.**
All of them

**What is the target framework and operating system affected by this issue? Please see [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [x] netCore 2.0
- [x] netCore 1.0
- [x] 4.7
- [x] 4.6.x
- [x] 4.5.x

**Please describe the current behaviour you are experiencing?**
In wrapping Serilog in one of my projects I stumbled upon the fact that the `ConsoleSink` class is not accessible from the outside. Now I personally like this, and would like to keep it hidden from the outside. But I would also like access to the resulting `ILogEventSink`. So maybe add a ConsoleSinkFactory that builds the `ILogEventSink`

```
var consoleSink = ConsoleSinkFactory.Create()
                                    .With(formatter)
                                    .With(standardErrorFromLevel)
                                    .With(ConsoleTheme.None)
                                    .Build();
```

**Comment by adamchester** [comment]
Hi @droosma, thanks for stopping by.

What exactly do you intend to do with the `ILogEventSink`? You’ve only said you want access to it, but not explained why.

**Comment by droosma** [comment]
I need to be able to construct / configure the Console sink in one place and basically `loggerConfiguration.WriteTo.Sink(eventSink);` somewhere else. This work for most Sinks. But your project [hides](https://github.com/serilog/serilog-sinks-console/blob/dev/src/Serilog.Sinks.Console/Sinks/SystemConsole/ConsoleSink.cs#L26) the `ILogEventSink`. I like the approach, but still need access to it, so thought this might be a solution

**Comment by nblumhardt** [comment]
Hi @droosma; thanks for the PR.

This is already possible, in a very obscure way, by using the `LoggerSinkConfiguration.Wrap()` method. You can use the delegate that it accepts to exfiltrate the `ILogEventSink` that's passed through it. Let me know if you need a hand getting it hooked up :-)

**Comment by droosma** [comment]
It's of course completely up to you guys. but for me the `LoggerSinkConfiguration.Wrap()` "hack" doesn't really fit. The fact that even you call it "obscure" might be a nice hint.

**Comment by nblumhardt** [comment]
It's not pretty, that's for sure 😊

But in the context of how it's used: Serilog isn't designed in such a way that sink types are expected to be public, the way some other logging libraries are. Over time it's been a decent design choice for the library, i.e. we've been able to make big changes to the internals of this package, and several others, that would have been messier if public sink types had to be kept backwards-compatible. (We've got some experience with this via _File_ and _PeriodicBatching_.)

The way Serilog works, we can keep backwards compatibility concerns wrapped up in the static sink configuration methods, which are mostly flat and fairly easy to reason about.

Even if this design has its own pros/cons, having to use a slightly roundabout approach to get at the underlying sink is a reasonable trade-off in comparison with keeping to the general intentions of the design.

Hope this explains the thinking in a bit more detail - thanks for all the input!
